templates/default/atlantis.yaml.erb

```diff
@@ -1,9 +1,5 @@
----
-<% @vars.each do |k,v| %>
-<% next if k == 'config' %>
-<% if v.is_a?(Array) %>
-<%= k %>: <%= v.join(',') %>
-<% next %>
-<% end %>
-<%= k %>: <%= if k == 'log-level' then v.downcase else v end %>
-<% end %>
+# yamllint disable-file
+# ##################### CHEF MANAGED DO NOT EDIT #########################
+# ###################### Atlantis Configuration ##########################
+
+<%= @vars.to_yaml %>
```

Comment:
I get wanting to keep this generic as that's how I originally wrote it, but then I found these gotchas and tried to remove some of these gotchas from the user problem, this removes that. I can't say I am a fan of that.

Reply:
i mean the only 'gotcha' was the log level in lowercase and automatically joining arrays with commas but i think its better to have the user be explicit about the data they want than to manipulate it in the background. The day atlantis has a value that is a yaml array the template gets broken anyway. (i do not know of any actual array values currently)

I favour keeping it simple and knowing that the hash input will be converted to yaml following what you would expect and having the user do any manipulations vs manipulating it too much but at the end of the day it is your hard work and i am just contributing some changes

Reply:
actually looking at our internal code and why we needed to change the template originally is the repo config uses arrays in yaml and the template kept joining them as strings making the syntax invalid.

https://www.runatlantis.io/docs/repo-level-atlantis-yaml.html#example-using-all-keys

that means that we cannot use the atlantis_config resource to add repo configs out of the box because of the fix for the gotchas in the original template which was designed only to handle server configurations

Reply:
Gotcha, I am not gonna lie I have not touched this code or the upstream project in quite a while. I wrote it and it served my personal and org use cases but clearly was fairly narrow in focus.This should still be considered a major/breaking change as we are changing what is acceptable as an input for a valid config.

Reply:
I have listed it as a Breaking changes in the change log and PR description including details of what changed, the major version is also being bumped. Anywhere else you think i should list it or things to do to make the breaking change more obvious?

Reply:
Nope we are good, I approved it. Sadly a lot of people do not understand what a breaking change really means so I always like making sure when I review I acknowledge the decision to accept a breaking change as well as confirmation that it will be bumped accordingly. I just had a major outage a couple weeks ago caused by a vendor I shall not name that did not understand this.